### PR TITLE
feat: add reversed mode & improvements to lyrics animation

### PR DESCRIPTION
**Pull request by GalvinGao**
## Reversed Mode
Allows the user to play music in a reversed order, opposite to the normal top-to-bottom direction. I found this feature plays quite nice with old playlists that I want to go back and have a taste with. With that saying, it is hidden by default behind a settings item located in the settings menu, to avoid confusion and clutter introduced to all users.

## Lyrics Animation
Inspired by Apple Music, I've tried to replicate the same effect by using `font-size` animation. On my MacBook Pro 2021 it looks quite smoothly and it looks astonishing with dynamic background. This is just to try and have some fun and it turns out to be quite nicely.

I also added a slight different styling to the translated lyrics, making it a bit transparent and a bit smaller to the original lyrics to distinguish the two.

ezgif-3-dcf06ca065

Also I've get rid of `v-html` by using `v-if` and `v-if-else`, just to make the code more readable and to get rid of extra DOM parsing.

### CODE REVIEW

File: src/views/lyrics.vue [hunk 1]
```diff
@@ -394,10 +404,8 @@ export default {
       }, 50);
     },
     formatLine(line) {
```
Comment (memorydream) [inline src/views/lyrics.vue:406]: remove it? i not found call

Comment (pan93412) [reply src/views/lyrics.vue:406]: Seems like it has been existed for a long time. Are you sure that it is unused anywhere?

File: src/utils/Player.js [hunk 2]
```diff
@@ -84,6 +85,18 @@ export default class {
       this._shuffleTheList();
     }
   }
+  get reversed() {
+    return this._reversed;
+  }
+  set reversed(reversed) {
+    if (this._isPersonalFM) return;
+    if (reversed !== true && reversed !== false) {
```
Comment (pan93412) [inline src/utils/Player.js:93]: 檢查類型建議用 typeof: `typeof reversed !== 'boolean`

File: src/utils/Player.js [hunk 3]
```diff
@@ -180,27 +193,43 @@ export default class {
     }, 1000);
   }
   _getNextTrack() {
+    const next = this._reversed ? this.current - 1 : this.current + 1;
```
Comment (pan93412) [inline src/utils/Player.js:196]: const next = this.current + (this._reversed ? -1 : 1);

個人拙見。

File: src/utils/Player.js [hunk 4]
```diff
@@ -180,27 +193,43 @@ export default class {
...
+
     if (this._playNextList.length > 0) {
       let trackID = this._playNextList.shift();
       return [trackID, this.current];
     }
 
-    // 当歌曲是列表最后一首 && 循环模式开启
-    if (this.list.length === this.current + 1 && this.repeatMode === 'on') {
-      return [this.list[0], 0];
+    // 循环模式开启，则重新播放当前模式下的相对的下一首
```
Comment (pan93412) [inline src/utils/Player.js:203]: 能有 Unit Test 更好。

**Review by pan93412 — CHANGES_REQUESTED** [review]

**Pull request merged**